src/auth/session/storage/postgresql.ts

```diff
@@ -0,0 +1,140 @@
+      ),
+      opts,
+    );
+  }
+
+  public readonly ready: Promise<void>;
+  private options: PostgreSQLSessionStorageOptions;
+  private client: pg.Client;
+
+  constructor(
+    private dbUrl: URL,
+    opts: Partial<PostgreSQLSessionStorageOptions> = {},
+  ) {
+    if (typeof this.dbUrl === 'string') {
+      this.dbUrl = new URL(this.dbUrl);
+    }
+    this.options = {...defaultPostgreSQLSessionStorageOptions, ...opts};
+    this.ready = this.init();
+  }
+
+  public async storeSession(session: SessionInterface): Promise<boolean> {
+    await this.ready;
+
+    const entries = sessionEntries(session);
+    const query = `
+      INSERT INTO ${this.options.sessionTableName}
+      (${entries.map(([key]) => key).join(', ')})
+      VALUES (${entries.map((_, i) => `$${i + 1}`).join(', ')})
+      ON CONFLICT (id) DO UPDATE SET ${entries
+        .map(([key]) => `${key} = Excluded.${key}`)
+        .join(', ')};
+    `;
+    await this.query(
+      query,
+      entries.map(([_key, value]) => value),
+    );
+    return true;
+  }
+
+  public async loadSession(id: string): Promise<SessionInterface | undefined> {
+    await this.ready;
+    const query = `
+      SELECT * FROM ${this.options.sessionTableName}
+      WHERE id = $1;
+    `;
+    const rows = await this.query(query, [id]);
+    if (!Array.isArray(rows) || rows?.length !== 1) return undefined;
+    const rawResult = rows[0] as any;
+    return sessionFromEntries(Object.entries(rawResult));
+  }
+
+  public async deleteSession(id: string): Promise<boolean> {
+    await this.ready;
+    const query = `
+      DELETE FROM ${this.options.sessionTableName}
+      WHERE id = $1;
+    `;
+    await this.query(query, [id]);
+    return true;
+  }
+
+  public disconnect(): Promise<void> {
+    return this.client.end();
+  }
+
+  private async init() {
+    this.client = new pg.Client({connectionString: this.dbUrl.toString()});
+    await this.connectClient();
+    await this.createTable();
+  }
+
+  private async connectClient(): Promise<void> {
+    await this.client.connect();
+  }
+
+  private async hasSessionTable(): Promise<boolean> {
+    const query = `
+      SELECT * FROM pg_catalog.pg_tables WHERE tablename = $1
+    `;
+    const [rows] = await this.query(query, [this.options.sessionTableName]);
+    return Array.isArray(rows) && rows.length === 1;
+  }
+
+  private async createTable() {
+    const hasSessionTable = await this.hasSessionTable();
+    if (!hasSessionTable && !this.options.createDBWhenMissing) {
+      throw Error('Session Table is missing');
+    } else if (!hasSessionTable) {
+      const query = `
+        CREATE TABLE ${this.options.sessionTableName} (
+          id varchar(255) NOT NULL PRIMARY KEY,
+          shop varchar(255) NOT NULL,
+          state varchar(255) NOT NULL,
+          isOnline boolean NOT NULL,
+          scope varchar(255),
+          expires varchar(255),
+          accessToken varchar(255),
+          onlineAccessInfo varchar(4096)
```

Comment:
based on the test, `onlineAccessInfo` would be PII data. I don't know if by default we want developers to save this information. Would be good to get the auth team to do a review of this and perhaps even @ragalie since he's helping lead the privacy governance mission

Reply:
I literally just ported whatever the in-memory storage was storing. What should I reduce this to? `id`, `shop`, `state`, `isOnline`, `scope` and `accessToken`?

Reply:
Agreed, it would be better not to store the profile information by default. Privacy Governance is currently focused on customer data, not staff member data, but the general principle of "minimize the data your app stores" holds in this situation too, I think.